### PR TITLE
PRO-5711 refactor login schema

### PR DESCRIPTION
**Pull request by triskuit**
## Summary
Update the login form to draw its schema from the user piece config instead of being hardcoded, standardizing the form verbiage. [PRO-5711](https://linear.app/apostrophecms/issue/PRO-5711/the-apostrophe-login-form-cannot-be-localized-or-customized-through)

## What are the specific steps to test this change?
**Acceptance Criteria**
As a developer, you can restate the username field with a different label at project level in project-level configuration of @apostrophecms/user and see that updated label in the login form, not just “manage users”

1. `npm link` to a test project
2. Override the label of the username field in `modules/@apostrophecms/user/index.js` (example below)
3. `npm run dev` => http://localhost:3000/login
4. Your test phrase is displayed in the username field

```
module.exports = {
  fields(self, options) {
    return {
      add: {
        username: {
          type: 'string',
          label: 'THIS IS A TEST',
          required: true,
          following: 'title'
        }
      }
    };
  }
};
```

## What kind of change does this PR introduce?
- [x] Bug fix
- [ ] New feature
- [x] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:
- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/login/index.js [hunk 1]
```diff
@@ -909,8 +910,20 @@ module.exports = {
...
-      }
+      },
 
+      getSchema() {
+        return self.apos.user.schema
+          .filter(({ name }) => [ 'username', 'password' ].includes(name))
+          .map(field => ({
+            name: field.name,
+            label: field.label,
+            placeholder: `Enter ${field.label.replace('apostrophe:', '').toLowerCase()}`,
```
Comment (triskuit) [inline modules/@apostrophecms/login/index.js:921]: We're passing in a phrase which usually gets parsed. I just stripped it but if there's a better way lemme know.

Comment (boutell) [reply modules/@apostrophecms/login/index.js:921]: Hmm. This is messy, isn't it. Our original goal here was to make this placeholder easy to override and localize server side, and now it it's server side, but it's still not possible to localize fully 😄 

So I suggest:

* Add `usernamePlaceholder` and `passwordPlaceholder` options to this module
* Default those to `apostrophe:enterUsername` and `apostrophe:enterPassword` and add the appropriate phrases to `en.json`
* Utilize those options as needed here as part of your massaging of the schema array

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/login/index.js [hunk 2]
```diff
@@ -66,6 +70,7 @@ module.exports = {
     }
   },
   async init(self) {
+    console.log({ opt: self.options });
```
Comment (boutell) [inline modules/@apostrophecms/login/index.js:73]: Stray log

File: modules/@apostrophecms/login/index.js [hunk 3]
```diff
@@ -909,8 +915,20 @@ module.exports = {
...
-      }
+      },
 
+      getSchema() {
+        return self.apos.user.schema
+          .filter(({ name }) => [ 'username', 'password' ].includes(name))
+          .map(field => ({
+            name: field.name,
+            label: field.label,
+            placeholder: field.placeholder ?? self.options.placeholder[field.name],
```
Comment (boutell) [inline modules/@apostrophecms/login/index.js:926]: Why the check for an existing placeholder from the user module? We definitely want one on the login form, so I think you can simplify.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/login/index.js (hunk 3 above)
Comment (triskuit) [reply modules/@apostrophecms/login/index.js:926]: It feels confusing to be able to override the label from the user module but not the placeholder, yeah? Or am I confusing something?

File: modules/@apostrophecms/login/index.js (hunk 2 above)
Comment (triskuit) [reply modules/@apostrophecms/login/index.js]: 🤦‍♀️

**Review by boutell — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**